src/librbd/io/ImageRequestWQ.cc

```diff
@@ -453,6 +472,10 @@ void ImageRequestWQ::process(ImageRequest<> *req) {
                  << "req=" << req << dendl;
 
   req->send();
+  if (m_image_ctx.get_max_iops()) {
```

Comment:
Probably should be doing this under the ```m_lock``` above to support the case when multiple IO dispatch threads are supported.